src/server.ts

```diff
@@ -201,10 +199,10 @@ export class AceBaseServer extends SimpleEventEmitter {
         addMetadataRoutes(routeEnv);
 
         // If environment is development, add API docs
-        if (process.env.NODE_ENV?.trim?.() === 'development') {
+        if (process.env.NODE_ENV === 'development') {
```

Comment:
Please keep the trim. When starting the server through the test script, the variable is suffixed with a single space. See https://github.com/appy-one/acebase-server/blob/183841756facfd2d1b58764bfa88bf92207a0ca9/package.json#L19

Reply:
Hmm... That's why it's there... TypeScript transpiles the optional chaining into such a mess, that no bundler can treeshake that. I'll try to think what could be done here...

Reply:
Can you try if changing the test script to `set NODE_ENV='development' && ...` fixes that?

Reply:
> That does not work, but `"test": "set NODE_ENV=development&& npm run start",` actually does. Not very pretty though...

@appy-one, what do you think? Is that ok, or shall we find a different solution?

Reply:
Nah, I've been on that route before and decided against it. Just keep the `trim` then.

Reply:
Made this for now. Let me know if there is a better solution.

Reply:
@Azarattum I really prefer keeping the trim. I prefer readability over a couple more bytes in generated code, any time

Reply:
It's not just a couple more bytes, it's literally x2 the bundle size. Which also means the build speed is 2 times slower. But this isn't even the primary issue. The whole motivation for this PR was that `swagger` literally breaks **the whole build** in bundled mode. Meaning you get errors before it even executes anything, so there's not even a chance to patch in in runtime. Have you seen the mess TypeScript creates from optical chaining? No wonder why bundlers can't parse that...

Reply:
There is actually a perfect solution for this whole issue, but it would mean some major changes... If the project had used `vite` to bundle the library instead of just `tsc`, we would have access to `import.meta.DEV` which works flawlessly. I could work on such a migration, but not sure if you are up to it in the first place...

Reply:
By the way, I honestly don't understand why the docs are there in the first place? Wouldn't it make more sense to have them separately hosted somewhere else (e.g. GitHub pages) for everyone to use? Bundling them with the library seems like a really weird approach.

Reply:
@appy-one Check out comments below [this post](https://dev.to/kabartolo/how-to-document-an-express-api-with-swagger-ui-and-jsdoc-50do). There is an instruction on how to generate a static docs page with `swagger`.

Reply:
> It's not just a couple more bytes, it's literally x2 the bundle size. Which also means the build speed is 2 times slower.

So this: `((_b = (_a = process.env.NODE_ENV) === null || _a === void 0 ? void 0 : _a.trim) === null || _b === void 0 ? void 0 : _b.call(_a)) === 'development'` is x2 the bundle size??

> Have you seen the mess TypeScript creates from optical chaining? No wonder why bundlers can't parse that...

It's mess needed to safely perform the same in < ES2020 code. Why don't you just set your build target to ES2020 or higher?

Reply:
@appy-one, no you didn't get it. The 2x bundle size comes from the whole `swagger` stuff being included. Which happens because the "mess" breaks tree shaking.

Reply:
Setting build target to ES2020 or higher would be needed from the `acebase-server` itself, since everything outside the library just uses already precompiled dist package.

Reply:
> The 2x bundle size comes from the whole `swagger` stuff being included. Which happens because the "mess" breaks tree shaking.

Aha, that wasn't clear.

> Setting build target to ES2020 or higher would be needed from the `acebase-server` itself, since everything outside the library just uses already precompiled dist package. 

Why does your bundler use the commonjs build? Code `import`ed from acebase-server should use the ESM build, see package.json

Reply:
I'm targeting `cjs` when bundling my application. I guess therefore `vite` thinks it's more appropriate to use the `cjs` version of the library.

Reply:
@appy-one I think I've found a perfect solution, by replacing the optional chaining with a manual check:
```ts
if (process.env.NODE_ENV && process.env.NODE_ENV.trim() === 'development') {
```

I was able to:
```ts
    replace({
      "process.env.NODE_ENV": "false",
    }),
```
in my `vite.config.js`. Which makes the first check fail and tree-shakes everything properly, while also keeping the `trim`!